presto-verifier/src/test/java/com/facebook/presto/verifier/rewrite/TestQueryRewriter.java

```diff
@@ -466,6 +512,71 @@ public void testRewriteFunctionCalls()
                         "       MIN(a) AS a\n" +
                         "    FROM test_table\n" +
                         ") x");
+
+        // Test rewriting window functions with partition and order derived from the original.
+        assertCreateTableAs(
+                queryRewriter.rewriteQuery(
+                        "SELECT\n" +
+                                "    FIRST_VALUE(a) OVER (\n" +
+                                "        PARTITION BY b\n" +
+                                "    )\n" +
+                                "FROM test_table",
+                        CONTROL).getQuery(),
+                "SELECT\n" +
+                        "    IF(\n" +
+                        "        MIN(a) OVER (\n" +
+                        "            PARTITION BY\n" +
+                        "                b\n" +
+                        "        ) IS NOT NULL,\n" +
+                        "        MIN(a) OVER (\n" +
+                        "            PARTITION BY\n" +
+                        "                b\n" +
+                        "        ),\n" +
+                        "        MAX(a) OVER (\n" +
+                        "            PARTITION BY\n" +
+                        "                b\n" +
+                        "        )\n" +
+                        "    )\n" +
+                        "FROM test_table");
+
+        // Test rewriting window functions with partition and order resolving.
+        assertCreateTableAs(
+                queryRewriter.rewriteQuery(
+                        "SELECT\n" +
+                                "    ROW_NUMBER() OVER (\n" +
+                                "        PARTITION BY a\n" +
+                                "        ORDER BY b DESC\n" +
+                                "    )\n" +
+                                "FROM test_table",
+                        CONTROL).getQuery(),
+                "SELECT\n" +
+                        "    ROW_NUMBER() OVER (\n" +
+                        "        PARTITION BY b\n" +
+                        "        ORDER BY b DESC\n" +
+                        "    )\n" +
+                        "FROM test_table");
+
+        // Test mapping of multiple substitutions and matching of literal arguments.
+        assertCreateTableAs(
+                queryRewriter.rewriteQuery(
+                        "SELECT\n" +
+                                "    APPROX_PERCENTILE(a, 0.95)\n" +
```

Comment:
Maybe add a test with both `APPROX_PERCENTILE(v)` and `APPROX_PERCENTILE(v, e)` present in a single query

Reply:
Good idea! Revised.